hw_5/pandas_script.py

```diff
@@ -0,0 +1,267 @@
+                                                                 'Name=5S_rRNA;product=5S ribosomal RNA': '5S'})
+# Merge two dataframe
+merged_data = rrana_dataframe.merge(alignment_dataframe, on=['chromosome'], suffixes=['_x', '_y'])
+# Select suitable options
+intersect_rrna_dataframe = merged_data.loc[(merged_data['start_y'] <= merged_data['start_x']) &
+                                           (merged_data['end_y'] >= merged_data['end_x'])]
+
+intersect_rrna_dataframe_modify = intersect_rrna_dataframe.drop('strand_y', inplace=True, axis=1)  # optionally
+
+# SECOND TASK
+
+# I. Preporation
+dif_expr = pd.read_csv('/content/diffexpr_data.tsv.gz', sep='\t')
+
+# II. Additional analysis
+
+
+# Function for dividing gene expression on 4 groups
+def significant(data_frame):
+    if data_frame.logFC < 0 and data_frame.log_pval < .95:
+        return 'Non-significantly downregelated'
+    elif data_frame.logFC < 0 and data_frame.log_pval > .95:
+        return 'Significantly downregelated'
+    elif data_frame.logFC > 0 and data_frame.log_pval > .95:
+        return 'Significantly upregelated'
+    else:
+        return 'Non-significantly upregelated'
+
+
+dif_expr['Significance'] = dif_expr.apply(significant, axis=1)
+
+# Limit determination
+logFC_max = dif_expr.logFC.max() + 1
+logFC_min = dif_expr.logFC.min() - 1
+logFC_limit = max(abs(logFC_min), logFC_max)
+
+# Select record gene
+significant_dataframe = dif_expr.query('log_pval >= .95')
+top2_downregulated = significant_dataframe.sort_values(by=['logFC']).head(2)
+top2_upregulated = significant_dataframe.sort_values(by=['logFC'], ascending=False).head(2)
+top2_together = pd.concat([top2_downregulated, top2_upregulated], axis=0)
+
+# Create lists for annotation
+record_genes = top2_together.Sample.tolist()
+record_logFC = top2_together.logFC.tolist()
+record_log_pval = top2_together.log_pval.tolist()
+
+# Change settings rcParam
+plt.rcParams['mathtext.fontset'] = 'dejavuserif'
+plt.rcParams['mathtext.bf'] = 'italic:bold'
+plt.rcParams['xtick.minor.visible'] = True
+plt.rcParams['ytick.minor.visible'] = True
+plt.rcParams['xtick.major.width']: 1
+plt.rcParams['xtick.major.size']: 10
+plt.rcParams['xtick.major.width']: 1
+plt.rcParams['xtick.major.size']: 10
+plt.rcParams['font.serif']: 'Times New Roman'
+
+# III. Graphs
+plt.figure(figsize=[10, 8], dpi=700)
+
+# Creation of scatter plot
+expr_graph = sns.scatterplot(data=dif_expr, x="logFC", y="log_pval",
+                             hue='Significance',
+                             hue_order=['Significantly downregelated', 'Significantly upregelated',
+                                        'Non-significantly downregelated', 'Non-significantly upregelated'],
+                             s=5, alpha=1, linewidth=0)
+
+# Create horizontal threshold and label
+expr_graph.axhline(y=0.95, color='grey', linewidth=1, linestyle='dashed')
```

Comment:
У тебя по оси y отложено -log10(p_value), а значит линия должна быть не 0.05, а -log10(0.05)
```suggestion
expr_graph.axhline(y=-np.log10(0.05), color='grey', linewidth=1, linestyle='dashed')
```